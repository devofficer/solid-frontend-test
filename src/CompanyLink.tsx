import { Company } from './types'

type Props = Company & {
  selected: boolean,
  onSelectCompany: (id: number) => void,
}

const CompanyLink = ({ id, name, selected, onSelectCompany }: Props) => (
  <div
    className={`nav__dropdown-menu-item company-item ${selected ? 'selected' : ''}`}
    onClick={() => onSelectCompany(id)}
  >
    <span>{name}</span>
    {selected && (
      <i className='material-icons'>
        check
      </i>
    )}
  </div>
)

export default CompanyLink
