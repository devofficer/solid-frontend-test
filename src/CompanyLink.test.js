import { shallow } from 'enzyme'
import CompanyLink from './CompanyLink'

describe('<CompanyLink />', () => {
  const setSelectedCompanyIdMock = jest.fn()

  const render = (props) => shallow(
    <CompanyLink
      onSelectCompany={setSelectedCompanyIdMock}
      {...props}
    />
  )

  it('renders link with selected state', () => {
    expect(render({ id: 1, name: 'company', selected: true })).toMatchSnapshot()
  })
})
