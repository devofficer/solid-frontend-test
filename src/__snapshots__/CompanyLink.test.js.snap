// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CompanyLink /> renders link with selected state 1`] = `
<div
  className="nav__dropdown-menu-item company-item selected"
  onClick={[Function]}
>
  <span>
    company
  </span>
  <i
    className="material-icons"
  >
    check
  </i>
</div>
`;
